src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,52 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A task that the user wants to complete by a certain point in time.
+ */
+public class Deadline extends Task {
+    protected boolean hasTime;
+    protected LocalDateTime dateTime;
+    protected LocalDate date;
+    protected DateTimeFormatter formatInput;
+    protected DateTimeFormatter formatOutputTime = DateTimeFormatter.ofPattern("MMM d yyyy hh:m a");
+    protected DateTimeFormatter formatOutputNoTime = DateTimeFormatter.ofPattern("MMM d yyyy");
+
+    /**
+     * Constructor for the Deadline class.
+     *
+     * @param description Description of the task.
+     * @param dateString  Date and time of the deadline.
+     * @param formatInput How the input is formatted.
+     * @param hasTime     Boolean for whether time was supplied.
+     */
+    public Deadline(String description, String dateString, DateTimeFormatter formatInput, boolean hasTime) {
+        super(description);
+        this.formatInput = formatInput;
+        this.hasTime = hasTime;
+        if (!hasTime) {
+            this.date = LocalDate.parse(dateString, formatInput);
+        } else {
+            this.dateTime = LocalDateTime.parse(dateString, formatInput);
+        }
+    }
+
+    /**
+     * Provides a String representation of the duke.task.Deadline.
+     *
+     * @return A String representation of the duke.task.Deadline.
+     */
+    @Override
+    public String toString() {
```

Comment:
Small nitpick, but overridden methods can get their javadocs copied from the parent class. The javadoc requires no substantial change unless the Overriden method does something completely different in the subclass.

Reply:
I see, that saves a whole bunch of effort as well. Thanks!